src/components/ImageGrid/ImageGridComponent.tsx

```diff
@@ -98,26 +105,48 @@ const ImageGridComponent = ({
     } else {
       setRowCol({ col: 1, row: 1 });
     }
-  }, [numberOfImgs, images]);
+  }, [numberOfImgs, randomPhotos]);
+
+  type unsplashPhotoFortmat = {
+    id: string;
+    url: string;
+    alt: string;
+    urls: { regular: string };
+    alt_description: string;
+  };
+
+  /*----------------------FETCH RANDOM IMAGES ---------------------- */
 
-  // to avoid bug undefined url when user change number of images
   useEffect(() => {
-    for (let i = 0; i < numberOfImgs; i++) {
-      images[i] = exampleImages[i];
+    async function getData() {
+      try {
+        const response = await axios.get(unsplashUrl);
+        const data = response.data; // this will return 10 results
+        const slicedArray = data
+          .slice(0, numberOfImgs)
+          .map(function (item: unsplashPhotoFortmat) {
+            return {
+              id: item.id,
+              url: item.urls.regular,
+              alt: item.alt_description,
+            };
+          });
+        setRandoPhotos(slicedArray);
+      } catch (error) {}
     }
+    getData();
   }, [numberOfImgs]);
-
   return (
     <ImageGrid
       height={imagesGridHeight}
-      width={imagesGridWidth}
+      maxWidth={imagesGridMaxWidth}
       numberOfImgs={numberOfImgs}
       row={rowCol.row}
       col={rowCol.col}
     >
-      {images.map((img, index) => (
+      {randomPhotos.map((photo: unsplashPhotoFortmat, index) => (
         <ImageWrap>
-          <ImageItem src={img} />
+          <ImageItem src={photo.url || ""} alt={photo?.alt || ""} />
```

Comment:
`src` should never be empty, if there's no url, you cannot render an image at all
`alt` should have some default at least